src/main/java/com/songoda/epiclevels/database/DataUpdater.java

```diff
@@ -0,0 +1,179 @@
+        super.onDisable();
+        toUpdate.clear();
+    }
+
+    public DataManager getManager() {
+        return manager;
+    }
+
+    @Override
+    public DatabaseConnector getConnector() {
+        return manager.getDatabaseConnector();
+    }
+
+    @Override
+    public String getTablePrefix() {
+        return manager.getTablePrefix();
+    }
+
+    public void sendUpdate(String id, Object... args) {
+        String[] strings = new String[args.length];
+        for (int i = 0; i < args.length; i++) {
+            strings[i] = String.valueOf(args[i]);
+        }
+        String msg = id + (args.length > 1 ? ":" + args.length : "") + "=" + String.join("_|||_", strings);
+        if (isEnabled()) {
+            Bukkit.getScheduler().runTaskAsynchronously(manager.getPlugin(), () -> sendMessage(msg));
+        }
+    }
+
+    public void sendPlayerUpdate(UUID uuid) {
+        sendUpdate("PLAYERUPDATE", uuid);
+    }
+
+    public void sendBoostCreate(UUID uuid, long duration, double multiplier, String sender) {
+        sendUpdate("BOOSTCREATE", uuid, duration, multiplier, sender);
+    }
+
+    public void sendBoostRemove(UUID uuid) {
+        sendUpdate("BOOSTREMOVE", uuid);
+    }
+
+    public void processMessage(String msg) {
+        String[] split = msg.split("=", 2);
+        if (split.length < 2) {
+            return;
+        }
+        String[] type = split[0].split(":");
+        if (type.length < 1) {
+            return;
+        }
+        String id = type[0];
+        int length = type.length > 1 ? Integer.parseInt(type[1]) : 1;
+
+        String[] args;
+        if (length <= 1) {
+            args = new String[] {split[1]};
+        } else {
+            args = split[1].split("_\\|\\|\\|_", length);
+            if (args.length != length) {
+                return;
+            }
+        }
+        try {
+            processUpdate(id, args);
+        } catch (Throwable t) {
+            t.printStackTrace();
+        }
+    }
+
+    public void processUpdate(String id, String[] args) {
+        switch (id.toUpperCase()) {
+            case "PLAYERUPDATE":
+                processPlayerUpdate(UUID.fromString(args[0]));
+                break;
+            case "BOOSTCREATE":
+                processBoostCreate(
+                        args[0].equalsIgnoreCase("null") ? null : UUID.fromString(args[0]),
+                        Long.parseLong(args[1]),
+                        Double.parseDouble(args[2]),
+                        args[3]
+                        );
+                break;
+            case "BOOSTREMOVE":
+                processBoostRemove(args[0].equalsIgnoreCase("null") ? null : UUID.fromString(args[0]));
+                break;
+            default:
+                throw new IllegalArgumentException("Cannot process update with the ID: " + id);
+        }
+    }
+
+    public void processPlayerUpdate(UUID uuid) {
+        if (toUpdate.contains(uuid)) {
+            return;
+        } else {
```

Comment:
If we return inside the if we can get rid of the `else`-Block for better readability